pkg/manager/impl/validation/project_validator.go

```diff
@@ -63,10 +63,17 @@ func ValidateProjectAndDomain(
 			"failed to validate that project [%s] and domain [%s] are registered, err: [%+v]",
 			projectID, domainID, err)
 	}
+
+	if project.State == nil {
+		return errors.NewFlyteAdminErrorf(codes.NotFound,
+			"project [%s] is not found", projectID)
+	}
+
```

Comment:
Trying to understand why we need this when the Get() returns an error when no records are found for the project
https://github.com/flyteorg/flyteadmin/blob/master/pkg/repositories/gormimpl/project_repo.go#L45

Reply:
hey @pmahindrakar-oss see this https://github.com/flyteorg/flyteadmin/pull/216 from the linked issue for an example of why we need this

Reply:
awesome find, thank you @pingsutw